x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/scalar/datetime/TimeParse.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.sql.expression.function.scalar.datetime;
+
+import org.elasticsearch.xpack.ql.expression.Expression;
+import org.elasticsearch.xpack.ql.expression.Expressions;
+import org.elasticsearch.xpack.ql.expression.function.scalar.BinaryScalarFunction;
+import org.elasticsearch.xpack.ql.expression.gen.pipeline.Pipe;
+import org.elasticsearch.xpack.ql.tree.NodeInfo;
+import org.elasticsearch.xpack.ql.tree.Source;
+import org.elasticsearch.xpack.ql.type.DataType;
+import org.elasticsearch.xpack.sql.type.SqlDataTypes;
+
+import java.time.ZoneId;
+
+import static org.elasticsearch.xpack.ql.expression.TypeResolutions.isString;
+import static org.elasticsearch.xpack.ql.type.DateUtils.UTC;
+import static org.elasticsearch.xpack.sql.expression.function.scalar.datetime.DateTimeParseProcessor.Parser.TIME;
+
+public class TimeParse extends BinaryDateTimeFunction {
+    
+    public TimeParse(Source source, Expression timestamp, Expression pattern) {
+        super(source, timestamp, pattern, UTC);
+    }
+
+    @Override
+    public DataType dataType() {
+        return SqlDataTypes.TIME;
+    }
+
+    protected TypeResolution resolveType() {
+        TypeResolution resolution = isString(left(), sourceText(), Expressions.ParamOrdinal.FIRST);
+        if (resolution.unresolved()) {
+            return resolution;
+        }
+        resolution = isString(right(), sourceText(), Expressions.ParamOrdinal.SECOND);
```

Comment:
Shouldn't the pattern be `isStringAndExact`? What do you think, @matriv ? Same question for timestamp.
One concerning aspect, especially for `pattern` is that if the user provides a field as a pattern and that field is `text` (vs. `keyword`) any text in it will be analyzed. Think of capital letters being lowercased with a `standard` analyzer. Should we be concerned about this?

Reply:
I understand the concern, but we don't check for exact (keyword only) in a various of other cases, like DATE_TRUNC, DATE_PART, etc.

Reply:
Maybe we should. Haven't given a lot of thought to something else other than the `pattern` but a `text` field can lowercase letters in a pattern and change the meaning of that pattern.

Reply:
I'd say to keep it with accepting also `text` for this PR.
We'll have to revisit all functions if we decide to exclude them and only accept exacts ('keyword'), and change the behavior consistently.